src/systems/explicit_system.C

```diff
@@ -45,6 +46,8 @@ ExplicitSystem::~ExplicitSystem ()
 {
   // clear data
   this->clear();
+
+  remove_vector("RHS Vector");
```

Comment:
Not sure, because System::~System calls this->clear(), which is ExplicitSystem::clear(), right? If so, rhs vector is not cleared.